src/main/java/com/budgeteers/financetracker/services/entry/EntryService.java

```diff
@@ -12,13 +13,13 @@ public class EntryService {
     private final List<IncomeEntry> incomeEntries = new ArrayList<>();  // TODO: Migrate to database for persistence
     private final List<ExpenseEntry> expenseEntries = new ArrayList<>();  // TODO: Migrate to database for persistence
 
-    public void addIncomeEntry(int amount) {
-        IncomeEntry entry = new IncomeEntry(amount);
+    public void addIncomeEntry(int amount, String incomeCategory) {
+        IncomeEntry entry = new IncomeEntry(amount, IncomeEntry.IncomeCategory.valueOf(incomeCategory));
```

Comment:
What happens if the user does not provide a valid category? In this case, you should catch the exception thrown by this line and respond with a status 400 and an explanation as to why it is a bad request. 

This also applies for the expense line 22.